pkg/kubectl/cmd/completion/completion.go

```diff
@@ -312,3 +317,51 @@ _complete kubectl 2>/dev/null
 	out.Write([]byte(zshTail))
 	return nil
 }
+
+// addPluginCommand adds plugin commands under the given command so that
+// completion includes them
+func addPluginCommands(kubectl *cobra.Command) {
+	output := &bytes.Buffer{}
+	streams := genericclioptions.IOStreams{
+		In:     &bytes.Buffer{},
+		Out:    output,
+		ErrOut: ioutil.Discard,
+	}
+
+	o := &plugin.PluginListOptions{
+		IOStreams: streams,
+		NameOnly:  true,
+	}
+	o.Complete(kubectl)
+	o.Run()
```

Comment:
Instead of doing this, which is hacky way of reading the plugin list, please change the plugin command so that it returns a list of plugins. Iow. it should have `ListPlugins` method that will return a list and its `Run` will just iterate over that list to print it. But your code will then use `ListPlugins` instead.

Reply:
OK, will do!

Reply:
@soltysh I changed the code as using `ListPlugins` instead, and squashed changes into a single commit.